src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+import net.dries007.tfc.common.capabilities.heat.HeatCapability;
+import net.dries007.tfc.common.recipes.BarrelRecipe;
+import net.dries007.tfc.common.recipes.BloomeryRecipe;
+import net.dries007.tfc.common.recipes.SoupPotRecipe;
+import net.dries007.tfc.config.TFCConfig;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.LampFuel;
+import net.dries007.tfc.util.calendar.Calendars;
+import net.dries007.tfc.util.calendar.ICalendar;
+
+/**
+ * Common tooltips that can be displayed for various block entities via external sources.
+ */
+public final class BlockEntityTooltips
+{
+    public static final BlockEntityTooltip BARREL = (level, state, entity, tooltip) -> {
+        if (state.getBlock() instanceof BarrelBlock && entity instanceof BarrelBlockEntity barrel)
+        {
+            if (state.getValue(BarrelBlock.SEALED))
+            {
+                BarrelRecipe recipe = barrel.getRecipe();
+                if (recipe != null)
+                {
+                    tooltip.accept(recipe.getTranslationComponent());
+                    tooltip.accept(Helpers.translatable("tfc.jade.sealed_date").append(ICalendar.getTimeAndDate(Calendars.CLIENT.ticksToCalendarTicks(barrel.getSealedTick()), Calendars.get(level).getCalendarDaysInMonth())));
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BELLOWS = (level, state, entity, tooltip) -> {
+        if (entity instanceof BellowsBlockEntity bellows)
+        {
+            int pushTicks = bellows.getTicksSincePushed();
+            if (pushTicks < 20)
+            {
+                pushTicks *= 2;
+                if (pushTicks > 20)
+                {
+                    pushTicks = 40 - pushTicks;
+                }
+
+                MutableComponent component = Helpers.translatable("tfc.jade.bellows_start");
+                String middle = Helpers.translatable("tfc.jade.bellows_middle").getString(); // probably better to translate this once.
+                for (int i = 0; i < pushTicks; i++)
+                {
+                    component.append(Helpers.literal(middle));
+                }
+                component.append(Helpers.translatable("tfc.jade.bellows_end"));
+                tooltip.accept(component);
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BLAST_FURNACE = (level, state, entity, tooltip) -> {
+        if (entity instanceof BlastFurnaceBlockEntity furnace)
+        {
+            furnace.getCapability(HeatCapability.BLOCK_CAPABILITY).ifPresent(cap -> heat(tooltip, cap.getTemperature()));
+
+            tooltip.accept(Helpers.translatable("tfc.jade.input_stacks", String.valueOf(furnace.getInputCount())));
+            tooltip.accept(Helpers.translatable("tfc.jade.catalyst_stacks", String.valueOf(furnace.getCatalystCount())));
+            tooltip.accept(Helpers.translatable("tfc.jade.fuel_stacks", String.valueOf(furnace.getFuelCount())));
```

Comment:
The `String.valueOf()` shouldn't be necessary here?
```suggestion
            tooltip.accept(Helpers.translatable("tfc.jade.input_stacks", furnace.getInputCount()));
            tooltip.accept(Helpers.translatable("tfc.jade.catalyst_stacks", furnace.getCatalystCount()));
            tooltip.accept(Helpers.translatable("tfc.jade.fuel_stacks", furnace.getFuelCount()));
```